### ISSUE TITLE
Global embedding should be cls token

### ISSUE THREAD

**Issue by jano1906** [szczurek-lab/jointformer#33]
https://github.com/szczurek-lab/jointformer/blob/910239b97a7b2a44c6e6fb6b320c244eea5e0396/jointformer/models/utils.py#L71

**Comment by jano1906** [comment]
`        if _is_causal:
            if next_token_only:
                outputs["logits_generation"] = self.lm_head(outputs['embeddings'][:, [-1], :])
            else:
                outputs["logits_generation"] = self.lm_head(outputs["embeddings"])
        else:
            outputs["logits_physchem"] = self.physchem_head(outputs['embeddings'][:, 0, :])
            outputs["logits_prediction"] = self.prediction_head(outputs['embeddings'][:, 0, :])


        return ModelOutput(
            attention_mask=attention_mask,
            embeddings=outputs['embeddings'],
            logits_generation=outputs.get('logits_generation', None),
            logits_physchem=outputs.get('logits_physchem', None),
            logits_prediction=outputs.get('logits_prediction', None),
            loss=None
        )
`

https://github.com/szczurek-lab/jointformer/blob/910239b97a7b2a44c6e6fb6b320c244eea5e0396/jointformer/models/jointformer.py#L92-L109

The interpretation of embeddings changes based on task. It is not clear how to interpret the first token embedding not knowing if model was in generation or prediction mode. Model should output distinct entry "cls" if it is in prediction mode and this entry should be None otherwise.
